app/code/core/Mage/Core/Model/Resource/Db/Abstract.php

```diff
@@ -400,6 +400,10 @@ public function load(Mage_Core_Model_Abstract $object, $value, $field = null)
      */
     protected function _getLoadSelect($field, $value, $object)
     {
+        $fields = $this->_getReadAdapter()->describeTable($this->getMainTable());
+        if(isset($fields[$field]['DATA_TYPE']))   {
```

Comment:
Nit-pick, but the check for `['DATA_TYPE']` is already performed within prepareColumnValue so is redundant to do here.

Reply:
You're very right, I hadn't actually checked prepareColumnValue too well but actually saw that this morning when I put it live. Unfortunately we're still seeing queries for increment_id without parentheses so my quest continues.

Reply:
@pepijnblom a side question - how do you monitor such a queries? Is there a generic way to log all queries using wrong data types?

Reply:
@tmotyl Not that I know of. In this case I noticed it because the queries were stacking up during peak hours and they were showing up in the mysql slow log because they were not using indexes. It's one thing to see a Magento query with 10 joins showing up in there, but you know something's wrong when it's a single select * from table :)

I found the other location causing the problem. It's in:
app/code/core/Mage/Sales/Model/Resource/Quote.php

I'll make a pull request for that too.

Reply:
@pepijnblom will you take care of the comment from @colinmollenhour  ?